plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelsAction.java

```diff
@@ -93,27 +107,48 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLUnde
         String[] modelIds = undeployModelsRequest.getModelIds();
         String[] targetNodeIds = undeployModelsRequest.getNodeIds();
 
-        if (modelAccessControlHelper.isModelAccessControlEnabled()) {
-            // Only allow user undeploy one model if model access control enabled.
-            if (modelIds == null || modelIds.length != 1) {
-                throw new IllegalArgumentException("only support undeploy one model");
-            }
-
+        if (modelIds.length == 1) {
```

Comment:
when there is only one model, why there is no need to check whether it's hidden model or not?

Reply:
We do check it. All validations are done [here](https://github.com/opensearch-project/ml-commons/pull/1981/files#diff-8edb43a502aa8b2462127ccfe0a9a89cbd298123526557155a3547d0dea2dd43L124) when there is only one model